packages/next/trace/report/to-json.ts

```diff
@@ -3,11 +3,68 @@ import { batcher } from './to-zipkin'
 import { traceGlobals } from '../shared'
 import fs from 'fs'
 import path from 'path'
+import { PHASE_DEVELOPMENT_SERVER } from '../../shared/lib/constants'
 
-let writeStream: fs.WriteStream
+let writeStream: RotatingWriteStream
 let traceId: string
 let batch: ReturnType<typeof batcher> | undefined
 
+const writeStreamOptions = {
+  flags: 'a',
+  encoding: 'utf8',
+}
+class RotatingWriteStream {
+  file: string
+  writeStream!: fs.WriteStream
+  size: number
+  sizeLimit: number
+  isRotating: Promise<void> | undefined
+  constructor(file: string, sizeLimit: number) {
+    this.file = file
+    this.size = 0
+    this.sizeLimit = sizeLimit
+    this.createWriteStream()
+  }
+  private createWriteStream() {
+    this.writeStream = fs.createWriteStream(this.file, writeStreamOptions)
+  }
+  // Recreate the file
+  private rotate(): void {
+    this.end()
+    try {
+      fs.unlinkSync(this.file)
+    } catch (err: any) {
+      // It's fine if the file does not exist yet
+      if (err.code !== 'ENOENT') {
+        throw err
+      }
+    }
+    this.size = 0
+    this.createWriteStream()
+  }
+  async write(data: string): Promise<void> {
+    this.size += data.length
+
+    if (this.size > this.sizeLimit) {
+      this.rotate()
+    }
+
+    await new Promise<void>((resolve, reject) => {
+      this.writeStream.write(data, 'utf8', (err) => {
+        if (err) {
+          return reject(err)
+        }
+
+        resolve()
+      })
+    })
```

Comment:
```suggestion
    if (!this.writeStream.write(data, 'utf8')) {
      await new Promise<void>((resolve, reject) => {
        this.writeStream.once('drain', resolve);
      })
    }
```

see https://nodejs.org/api/stream.html#stream_writable_write_chunk_encoding_callback